src/main/java/hudson/plugins/scm/koji/model/RPM.java

```diff
@@ -86,6 +92,29 @@ public boolean hasUrl() {
         return url != null;
     }
 
+    public void setHashSum(File file) throws IOException, NoSuchAlgorithmException {
+        if (file == null) {
+            hashSum = null;
+            return;
+        }
+        byte[] buffer = new byte[(int) file.length()];
```

Comment:
Seriously? Buffer of size of a file? What about 1gb files?
This would need fixing,  Please rework this method to be little bit more HW friendly.